.github/workflows/dapr-perf-components.yml

```diff
@@ -0,0 +1,544 @@
+#
+# Copyright 2021 The Dapr Authors
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Required secrets:
+# - AZURE_CREDENTIALS: JSON object containing the Azure service principal credentials. Docs: https://github.com/Azure/login#configure-a-service-principal-with-a-secret
+# - PERF_AZURE_STORAGE_ACCOUNT and PERF_AZURE_STORAGE_KEY: Credentials for the Storage Account where to store the result of perf tests
+# - DAPR_BOT_TOKEN: Token for the Dapr bot
+#
+# Optional secrets:
+# - AZURE_DIAG_LOG_ANALYTICS_WORKSPACE_ID: Resource ID of the Log Analytics Workspace where to store certain diagnostic logs (e.g. `/subscriptions/<subscription>/resourcegroups/<resource group>/providers/Microsoft.OperationalInsights/workspaces/<workspace name>`)
+# - AZURE_DIAG_STORAGE_ID: Resource ID of the Azure Storage account where to store certain diagnostic logs (e.g. `/subscriptions/<subscription>/resourcegroups/<resource group>/providers/Microsoft.Storage/storageAccounts/<storage account name>`)
+
+name: dapr-perf-components
+
+on:
+  # Run every Saturday at 4.34am UTC
+  schedule:
+    - cron: "34 4 * * 6"  # Manual trigger
+  workflow_dispatch:
+  # Dispatch on external events
+  repository_dispatch:
+    types: [components-perf-test]
+env:
+  # Configure proxy for Go modules
+  GOPROXY: https://proxy.golang.org
+  # Version of kubectl
+  KUBECTLVER: "v1.25.2"
+  # Version of Helm
+  HELMVER: "v3.10.0"
+  # Kubernetes namespace to use
+  DAPR_NAMESPACE: "dapr-tests"
+  # Timeout for tests
+  MAX_TEST_TIMEOUT: 5400
+  # Space-separated of supported Azure regions: one will be picked randomly for each cluster
+  AZURE_REGIONS: "westus3"
+  # Container registry where to cache perf test images
+  DAPR_CACHE_REGISTRY: "dapre2eacr.azurecr.io"
+
+jobs:
+  deploy-infrastructure:
+    name: Deploy test infrastructure
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up for scheduled test
+        if: github.event_name != 'repository_dispatch'
+        run: |
+          echo "CHECKOUT_REPO=${{ github.repository }}" >> $GITHUB_ENV
+          echo "CHECKOUT_REF=refs/heads/master" >> $GITHUB_ENV
+        shell: bash
+      - name: Parse test payload
+        if: github.event_name == 'repository_dispatch'
+        uses: actions/github-script@v6.2.0
+        with:
+          github-token: ${{secrets.DAPR_BOT_TOKEN}}
+          script: |
+            const testPayload = context.payload.client_payload;
+            if (testPayload && testPayload.command == "ok-to-perf") {
```

Comment:
```suggestion
            if (testPayload && testPayload.command == "ok-to-perf-components") {
```

Reply:
Later another PR can be added to change dapr-bot to send out the right repo dispatch on issuing the comment `/ok-to-perf-components`